src/ddmd/mtype.d

```diff
@@ -3075,6 +3075,13 @@ extern (C++) abstract class Type : RootObject
     {
         v.visit(this);
     }
+
+    final TypeFunction toTypeFunction()
+    {
+        if (ty != Tfunction)
+            assert(0);
```

Comment:
Since this should never trigger in a release-build
make it `assert(ty == Tfunction, "toTypeFunction only works on functions");`